### PR TITLE
Implement proxies for clientwebsocket.

### PR DESCRIPTION
**Pull request by Priya91**
## Description

With this PR, before attempting to open a connection to the host, if there is a proxy specified, which is not the DefaultWebProxy, in ClientWebSocketOptions, then this change tries to establish the connection via the proxy. 

1. Proxy servers that support basic and digest auth are supported. 
2. If the proxy wants to send a response body then that is also read from the stream, only when the content-length header is specified. Chunked content from proxies is not supported.

Also upgrades the projects to remove deprecated project systems.

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [ ] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [ ] If applicable, the public code is properly documented.
- [ ] Pull request includes test coverage for the included changes.
- [ ] The code builds without any errors.

### CODE REVIEW

File: src/Microsoft.Azure.Relay/WebSockets/NetCore/WebSocketHandle.Managed.cs [hunk 1]
```diff
@@ -197,6 +220,226 @@ public void Abort()
...
+        private async Task ConnectWithProxyAsync(Uri uri, ClientWebSocketOptions options, Stream stream, CancellationToken cancellationToken)
+        {
+            const string successStatusCode = "HTTP/1.1 200";
+            const string proxyAuthenticationStatusCode = "HTTP/1.1 407";
+            const string Basic = "Basic ";
+            const string Digest = "Digest ";
+            AuthenticationHeaderValue ahv = null;
+
+            string statusLine = await SendRequestToProxyAsync(uri, options, stream, ahv, cancellationToken);
+            if (statusLine == successStatusCode)
```
Comment (dlstucki) [inline src/Microsoft.Azure.Relay/WebSockets/NetCore/WebSocketHandle.Managed.cs]: 1.0 proxies are not supported, correct?

File: src/Microsoft.Azure.Relay/WebSockets/NetCore/WebSocketHandle.Managed.cs [hunk 2]
```diff
@@ -197,6 +220,226 @@ public void Abort()
...
+            {
+                // Proxy connection succeeded.
+                return;
+            }
+
+            // Proxy requires authentication.
+            if (statusLine.StartsWith(proxyAuthenticationStatusCode))
+            {
+                if (statusLine.Length > proxyAuthenticationStatusCode.Length &&
```
Comment (dlstucki) [inline src/Microsoft.Azure.Relay/WebSockets/NetCore/WebSocketHandle.Managed.cs]: Would it be easier to just change `proxyAuthenticationStatusCode` to have a space at the end?

File: src/Microsoft.Azure.Relay/WebSockets/NetCore/WebSocketHandle.Managed.cs [hunk 3]
```diff
@@ -80,16 +91,28 @@ public void Abort()
...
             CancellationTokenRegistration registration = cancellationToken.Register(s => ((WebSocketHandle)s).Abort(), this);
             try
             {
-                // Connect to the remote server
-                Socket connectedSocket = await ConnectSocketAsync(uri.Host, uri.Port, cancellationToken).ConfigureAwait(false);
-                Stream stream = new AsyncEventArgsNetworkStream(connectedSocket);
+
+                Socket connectedSocket;
+                Stream stream;
+                if (options.Proxy != null && options.Proxy != WebRequest.DefaultWebProxy)
```
Comment (dlstucki) [inline src/Microsoft.Azure.Relay/WebSockets/NetCore/WebSocketHandle.Managed.cs]: If there's a proxy set via config(?) or IE settings why not try to use it?

Comment (Priya91) [reply src/Microsoft.Azure.Relay/WebSockets/NetCore/WebSocketHandle.Managed.cs]: This requires interop native calls, and not top of our priority.

**Comment by dlstucki** [comment]
I submitted the C# changes in PR https://github.com/Azure/azure-relay-dotnet/pull/63

**Pull request closed without merging**